src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -525,11 +532,63 @@ class ContentSidebar extends PureComponent<Props, State> {
      * File tasks fetch success callback
      *
      * @private
-     * @param {Object} tasks - Box task
+     * @param {Object} tasks - Box task without assignments
+     * @return {void}
+     */
+    fetchTasksSuccessCallback = (tasksWithoutAssignments: Tasks): void => {
+        this.setState({ tasksWithoutAssignments, tasksError: undefined });
+        this.fetchTaskAssignments(tasksWithoutAssignments);
+    };
+
+    /**
+     * Update Tasks to include task assignments
+     *
+     * @private
+     * @param {Array<TaskAssignment>} entries - Box task assignment entries
+     * @param {TaskAssignments} assignments - Box task assigments
+     * @return {TaskAssignments} Updated Box task assignments
+     */
+    populateTaskAssignments(entries: Array<TaskAssignment>, assignments: TaskAssignments): TaskAssignments {
+        return {
+            total_count: assignments.entries.length,
+            entries: entries.map((item: TaskAssignment) => {
+                const assignment = assignments.entries.find((a) => a.id === item.id);
+                if (assignment) {
+                    return {
+                        ...assignment
+                    };
+                }
+                return item;
+            })
+        };
+    }
+
+    /**
+     * File task assignment fetch success callback
+     *
+     * @private
+     * @param {TaskAssignments} assignments - Box task assigment
      * @return {void}
      */
-    fetchTasksSuccessCallback = (tasks: Tasks): void => {
-        this.setState({ tasks, tasksError: undefined });
+    fetchTaskAssignmentsSuccessCallback = (assignments: TaskAssignments): void => {
+        const { tasksWithoutAssignments } = this.state;
+        if (!tasksWithoutAssignments) {
+            return;
+        }
+
+        const { entries, total_count } = tasksWithoutAssignments;
+        this.setState({
+            tasksWithoutAssignments: {
```

Comment:
should this, instead, be just `tasks`? Id assume that once all tasks had assignments, we could finally render in the AF

Reply:
Ah, I see down below you set via `tasks: this.state.tasksWithoutAssignments`

Reply:
That method updates the assignments for only that particular task so eventually when all those promises are resolved, we can update tasks